### DISCUSSION TITLE
trivy offline-db

### DISCUSSION THREAD

**Discussion by senemamourr** [aquasecurity/trivy#4838] [Q&A] [answered]
### Description

we want to refresh trivy db on no internet zone.
we download the offline db :
https://github.com/aquasecurity/trivy-db/releases/download/v1-2023020812/trivy-offline.db.tgz
unfortunately it does not work and got this error:
ERROR The local DB has an old schema version which is not supported by the current version of Trivy CLI. DB needs to be updated.
2023-07-18T11:10:38.694+0200 FATAL database error: validate error: --skip-update cannot be specified with the old DB schema. Local DB: 1, Expected: 2

is there a compatible db-offline with the latest version of trivy ?

### Desired Behavior

update the trivy database manually on air gapped environnement

### Actual Behavior

ERROR The local DB has an old schema version which is not supported by the current version of Trivy CLI. DB needs to be updated.
ERROR The local DB has an old schema version which is not supported by the current version of Trivy CLI. DB needs to be updated.

### Reproduction Steps

```bash
1.download offline-db from https://github.com/aquasecurity/trivy-db/releases/download/v1-2023020812/trivy-offline.db.tgz
2.tar xzvf trivy-offline.db.tgz
3. mv trivy.db .cache/trivy/db/
4. mv metadata.json .cache/trivy/db/
5. ./trivy server  --skip-db-update --listen 0.0.0.0:12000
...
```

### Target

None

### Scanner

None

### Output Format

None

### Mode

None

### Debug Output

```bash
2023-07-18T11:21:07.033+0200    DEBUG   cache dir:  /home/ansible/.cache/trivy
2023-07-18T11:21:07.036+0200    DEBUG   Skipping DB update...
2023-07-18T11:21:07.037+0200    ERROR   The local DB has an old schema version which is not supported by the current version of Trivy CLI. DB needs to be updated.
2023-07-18T11:21:07.040+0200    FATAL   database error:
    github.com/aquasecurity/trivy/pkg/commands/operation.DownloadDB
        /home/runner/work/trivy/trivy/pkg/commands/operation/operation.go:120
  - validate error:
    github.com/aquasecurity/trivy/pkg/db.(*Client).NeedsUpdate
        /home/runner/work/trivy/trivy/pkg/db/db.go:110
  - --skip-update cannot be specified with the old DB schema. Local DB: 1, Expected: 2:
    github.com/aquasecurity/trivy/pkg/db.(*Client).validate
        /home/runner/work/trivy/trivy/pkg/db/db.go:126
```

### Operating System

debian

### Version

```bash
Version: 0.43.1
Vulnerability DB:
  Version: 1
  UpdatedAt: 2023-02-08 12:48:16.989777838 +0000 UTC
  NextUpdate: 2023-02-08 18:48:16.989777438 +0000 UTC
  DownloadedAt: 0001-01-01 00:00:00 +0000 UTC
```

### Checklist

- [x] Run `trivy image --reset`
- [ ] Read [the troubleshooting](https://aquasecurity.github.io/trivy/latest/docs/references/troubleshooting/)

**Comment by DmitriyLewen** [comment, marked as answer]
Hello @senemamourr 
Thanks for your interest to Trivy!

You need to use `oras` to download Trivy-db - https://aquasecurity.github.io/trivy/v0.43/docs/advanced/air-gap/#download-the-vulnerability-database

**Reply by knqyf263** [reply]
Or `--download-db-only` flag as described in the document.
